### ISSUE TITLE
Other sensors and actuators

### ISSUE THREAD

**Issue by alexkoel1958** [jbouwh/ha-elro-connects#21]
Hi,

After reading the manual better I succeeded in installing the integration. It works great with a Elro K1 (found a refurbised one with Elro). As far as I understand it is only suitable for Fire/smoke, heat and CO alarm. Are you planning other sensors in the future? Elro has a nice priced set of addons. I'm interested in the temperature/humidity (SF40TH) and the radiator thermostat (SF40RV). The temperature sensor is great for monitoring or temperature regulation.

Regards
Alex

**Comment by jbouwh** [comment]
All alarm types should work, and also door/window sensors.
Other devices should work as well but will only return a state, a signal strength and battery sensor based on the current code.

**Comment by alexkoel1958** [comment]
So no plans for sensors returning a value like temperature? At he moment I'm using TFA sensors, but the range of those sensors is bad. It seems that the K1 has better specs.

**Comment by alexkoel1958** [comment]
How to use this debug lib. Do I need a running Linux system with python for it?

**Comment by jbouwh** [comment]
No, it is possible to use it on Windows. It might be a challenge though to discover where the files are installed.
You need Python 3.10 or higher.

**Comment by jbouwh** [comment]
Any progress with debugging?

**Comment by alexkoel1958** [comment]
No, Sorry. I have no clue how to get the debugging working. I have to do it in windows. I installed Phyton and did "pip install lib-elro-connects". This seems to have done something. After that I'm lost. I could't figure out how to do the debugging. 
I love to give a contribution, but I'm missing the Phyton skills I'm afraid.

**Comment by jbouwh** [comment]
After installing lib-elro-connects with `pip install lib-elro-connects` you can find the installed example scrips as follows:
- Open a `cmd` (command box), standard it will open in your home dir.
- Search for `lib-elro-connects` using: `dir /s lib-elro-connects`
- Something like ` Directory of C:\Users\user\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\share`
- Copy the found folder name and change directory:
- `cd C:\Users\user\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\share`
- `cd lib-elro-connects`
- `dir` shows 2 example scripts.

**Comment by JoSchaap** [comment]
> After installing lib-elro-connects with `pip install lib-elro-connects` you can find the installed example scrips as follows:
> 
> * Open a `cmd` (command box), standard it will open in your home dir.
> * Search for `lib-elro-connects` using: `dir /s lib-elro-connects`
> * Something like ` Directory of C:\Users\user\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\share`
> * Copy the found folder name and change directory:
> * `cd C:\Users\user\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\share`
> * `cd lib-elro-connects`
> * `dir` shows 2 example scripts.

Thanks for this, I actually also have one of these temp sensors; so I'll also give it a go :)
-merely replying to bookmark this for myself ^_^

**Comment by jbouwh** [comment]
Any progress here?

**Comment by Skons** [comment]
> Hi,
> 
> After reading the manual better I succeeded in installing the integration. It works great with a Elro K1 (found a refurbised one with Elro). As far as I understand it is only suitable for Fire/smoke, heat and CO alarm. Are you planning other sensors in the future? Elro has a nice priced set of addons. I'm interested in the temperature/humidity (SF40TH) and the radiator thermostat (SF40RV). The temperature sensor is great for monitoring or temperature regulation.
> 
> Regards Alex

All information is also available through the app that is as source available on github. All devices are ennumerated here
https://github.com/Siterwell/familywell-lidl-android/blob/6218fdd314a031c4745e71f101061065cc27bbfc/branch/lidl/app/src/main/java/me/hekr/sthome/tools/NameSolve.java

TH_CHECK looks like a candidate

If you'll search the repo for TH_CHECK you will find this
https://github.com/Siterwell/familywell-lidl-android/blob/6218fdd314a031c4745e71f101061065cc27bbfc/branch/lidl/app/src/main/java/me/hekr/sthome/model/modeladapter/GridAdapter.java#L104

That drawable.b11 probably resolves to this
https://github.com/Siterwell/familywell-lidl-android/blob/6218fdd314a031c4745e71f101061065cc27bbfc/branch/lidl/app/src/main/res/drawable/b11.png

Another search for TH_CHECK will bring you to battery and alerts management
https://github.com/Siterwell/familywell-lidl-android/blob/6218fdd314a031c4745e71f101061065cc27bbfc/branch/lidl/app/src/main/java/me/hekr/sthome/history/HistoryDataHandler.java#L149

The EquipmentAdapter, if i'm not mistaken, handles the information from the device. The link below parses the [device status update](https://github.com/jbouwh/lib-elro-connects/blob/main/protocol.md#device_status_update). There you can see how the the device state is parsed. There is a weird thing though with parsing the device state, all devices in the EquipmentAdapter handle eq.getState().substring(4,6), only the TH_CHECK parsed 2,4. In the battery management (HistoryDataHandler), substring 2,4 is already parsed, while 4,6 is used for alert state. So i think that is a bug in the app.
https://github.com/Siterwell/familywell-lidl-android/blob/6218fdd314a031c4745e71f101061065cc27bbfc/branch/lidl/app/src/main/java/me/hekr/sthome/equipment/adapter/EquipmentAdapter.java#L256

**Comment by jbouwh** [comment]
We could extend to that. In `lib-elro-connects` there is some example code that can help to find out if you can extract sensors. I successful added a smart plug that way (which i do not own myself).
